GlobalQuakeServer/src/main/java/gqserver/websocketserver/Clients.java

```diff
@@ -0,0 +1,149 @@
+package gqserver.websocketserver;
+
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.List;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+
+import org.tinylog.Logger;
+
+import globalquake.core.Settings;
+
+
+
+public class Clients {
+    private ScheduledExecutorService pingExecutor;
+
+    // IP:PORT -> Client
+    private Hashtable<String, Client> clients;
+    
+    //IP -> Integer
+    private Hashtable<String, Integer> uniqueIPConnectionCounts;
+    
+    private static Clients instance = new Clients();
+    private Clients() {
+        pingExecutor = Executors.newScheduledThreadPool(4);
+        clients = new Hashtable<String, Client>();
+        uniqueIPConnectionCounts = new Hashtable<String, Integer>();
+    }
+
+    public int getCountForIP(String ip) {
+        int count = 0;
+        try{
+            count = uniqueIPConnectionCounts.get(ip);
+        }
+        catch(Exception e) {}

```

Comment:
umm what? maybe you wanted to use HashMap instead of Hashtable.. there is a function called getOrDefault(...)